canmatrix/canmatrix.py

```diff
@@ -33,13 +33,16 @@
 from collections import OrderedDict
 
 import logging
+
+
 logger = logging.getLogger('root')
 try:
     import bitstruct
 except:
     bitstruct = None
     logger.info("bitstruct could not be imported // No signal de/encoding possible // try pip install bitstruct")
-#from past.builtins import basestring
+
+from past.builtins import basestring
```

Comment:
I would rather not allow mixing of byte-strings (py2 `str`) and unicode strings.  Do we really need to?  I do see that the use of basestring was actually added elsewhere.

http://python-future.org/compatible_idioms.html#basestring

I'm fine with this being considered/addressed separately from this PR.

Reply:
It is already in use in the code on line 396

Reply:
Right, that's why I don't intend to say it shouldn't be part of this change.  But, I am pointing out that I think it shouldn't be used at all.  @ebroecker, looks like you committed this.  Do we really want to allow py2 byte-strings to be considered strings?

https://github.com/ebroecker/canmatrix/commit/f949335a59e3bca70133e7809ea95a38d546dec3#diff-6e14cb4889f3ecdbe6c054170655f1b4R470

Reply:
not really - I even don't know anymore, what I wanted to check, if given 'value' is a string or a numeric value. I think it would be even more easy to check that given 'value' is numeric...

Reply:
Or just have a separate function that converts a string? Then you don't have to do any type checking in either. Both either work or don't.